### PR TITLE
acs-engine orchestrators should throw an error when an invalid/unsup…

### LINKED ISSUE #2852: acs-engine orchestrators should error when version doesn't exist
The acs-engine version command doesn't return unsupported when the provided version is not supported:

```
/go/src/github.com/Azure/acs-engine# bin/acs-engine orchestrators --orchestrator  kubernetes --version 1.222
{
  "orchestrators": [
    {
      "orchestratorType": "Kubernetes",
      "orchestratorVersion": "1.222"
    }
  ]
}
```

Should return a message similar to unsupported orchestrator for unsupported version:
```
/go/src/github.com/Azure/acs-engine# bin/acs-engine orchestrators --orchestrator stuff
Error: Unsupported orchestrator 'stuff'
```

Applies to all orchestrators.

### PR DESCRIPTION
**Pull request by tariq1890**
…ported version is provided

**What this PR does / why we need it**: The orchestrators command should error out when an invalid orchestrator version is provided as input.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #2852

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [x] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/api/orchestrators.go [hunk 1]
```diff
@@ -149,6 +149,19 @@ func kubernetesInfo(csOrch *OrchestratorProfile) ([]*OrchestratorVersionProfile,
 			return nil, fmt.Errorf("Kubernetes version %s is not supported", csOrch.OrchestratorVersion)
```
Comment (CecileRobertMichon) [inline pkg/api/orchestrators.go]: Is this code redundant now?

**Comment by tariq1890** [comment]
@CecileRobertMichon I've modified my PR as per your review comment :).

File: pkg/api/orchestrators.go [hunk 2]
```diff
@@ -22,6 +23,13 @@ func init() {
 		SwarmMode:  dockerceInfo,
 		OpenShift:  openShiftInfo,
 	}
+	versionsMap = map[string][]string{
```
Comment (CecileRobertMichon) [inline pkg/api/orchestrators.go]: even better :)

File: pkg/api/orchestrators.go (hunk 1 above)
Comment (CecileRobertMichon) [reply pkg/api/orchestrators.go]: @tariq1890 I think you can remove the check for `"<1.6.0"` since those will return false for !isVersionSupported(csOrch)

**Comment by tariq1890** [comment]
I've updated the PR.

**Pull request merged**